main.ts

```diff
@@ -267,7 +281,19 @@ export default class FilenameHeadingSyncPlugin extends Plugin {
         return {
           lineNumber: i,
           text: fileLines[i].substring(2),
+          style: HeadingStyle.Prefix,
         };
+      } else {
+        if (
+          fileLines[i + 1] !== undefined &&
+          fileLines[i + 1].match(/^=+$/) !== null
```

Comment:
This looks like it's hardcoding `=` and doesn't consider the setting for underline string

Reply:
Should it allow arbitrary strings for the underline then? Unless I'm confusing 2 things here